apps/server/src/modules/group/controller/api-test/group.api.spec.ts

```diff
@@ -255,4 +263,177 @@ describe('Group (API)', () => {
+
+				const syncedCourse: CourseEntity = courseFactory.build({
+					school,
+					syncedWithGroup: availableGroupInSchool,
+				});
+
+				await em.persistAndFlush([
+					adminAccount,
+					adminUser,
+					groupInSchool,
+					availableGroupInSchool,
+					groupInOtherSchool,
+					school,
+					otherSchool,
+					syncedCourse,
+				]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(adminAccount);
+
+				return {
+					loggedInClient,
+					groupInSchool,
+					availableGroupInSchool,
+				};
+			};
+			describe('when requesting all groups', () => {
+				it('should return all groups of the school', async () => {
+					const { loggedInClient, groupInSchool, availableGroupInSchool } = await setup();
+
+					const response = await loggedInClient.get();
+
+					expect(response.status).toEqual(HttpStatus.OK);
+					expect(response.body).toEqual([
+						{
+							id: groupInSchool.id,
+							name: groupInSchool.name,
+						},
+						{
+							id: availableGroupInSchool.id,
+							name: availableGroupInSchool.name,
+						},
+					]);
+				});
+			});
+
+			describe('when requesting all available groups', () => {
+				it('should return all available groups for course sync', async () => {
+					const { loggedInClient, availableGroupInSchool } = await setup();
+
+					const response = await loggedInClient.get('true');
+
+					expect(response.status).toEqual(HttpStatus.OK);
+					expect(response.body).toEqual([
+						{
+							id: availableGroupInSchool.id,
+							name: availableGroupInSchool.name,
+						},
+					]);
+				});
+			});
+		});
+
+		describe('when teacher requests groups', () => {
+			const setup = async () => {
+				const school: SchoolEntity = schoolEntityFactory.buildWithId();
+				const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher({ school });
+
+				const teachersGroup: GroupEntity = groupEntityFactory.buildWithId({
+					organization: school,
+					users: [{ user: teacherUser, role: teacherUser.roles[0] }],
+				});
+				const availableTeachersGroup: GroupEntity = groupEntityFactory.buildWithId({
+					organization: school,
+					users: [{ user: teacherUser, role: teacherUser.roles[0] }],
+				});
+				const groupWithoutTeacher: GroupEntity = groupEntityFactory.buildWithId({ organization: school });
+
+				const syncedCourse: CourseEntity = courseFactory.build({
+					school,
+					syncedWithGroup: availableTeachersGroup,
+				});
+
+				await em.persistAndFlush([
+					teacherAccount,
+					teacherUser,
+					teachersGroup,
+					availableTeachersGroup,
+					groupWithoutTeacher,
+					school,
+					syncedCourse,
+				]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(teacherAccount);
+
+				return {
+					loggedInClient,
+					teachersGroup,
+					availableTeachersGroup,
+				};
+			};
+			describe('when requesting all groups', () => {
+				it('should return all groups the teacher is part of', async () => {
```

Comment:
Please add one more group to the data. All test data you are creating the teacher is part of.. seems not to be a good test